public/src/main/java/com/revenuecat/purchases/models/SubscriptionOptions.kt

```diff
@@ -0,0 +1,104 @@
+package com.revenuecat.purchases.models
+
+import android.os.Parcelable
+import androidx.annotation.VisibleForTesting
+import kotlinx.parcelize.Parcelize
+
+@Parcelize
+class SubscriptionOptions(
+    private val subscriptionOptions: List<SubscriptionOption>
+    ) : List<SubscriptionOption> by subscriptionOptions, Parcelable {
+
+    /**
+     * Finds the base plan [SubscriptionOption].
+     */
+    val basePlan: SubscriptionOption?
+        get() = this.firstOrNull { it.isBasePlan }
+
+    /**
+     * Finds the first [SubscriptionOption] with a free trial [PricingPhase].
+     */
+    val freeTrial: SubscriptionOption?
+        get() = this.firstOrNull { it.freePhase != null }
+
+    /**
+     * Finds the first [SubscriptionOption] with an intro trial [PricingPhase].
+     * There can be a free trial [PricingPhase] and intro trial [PricingPhase] in the same [SubscriptionOption].
+     */
+    val introTrial: SubscriptionOption?
+        get() = this.firstOrNull { it.introPhase != null }
+
+    /**
+     * The default [SubscriptionOption]:
+     *   - Filters out offers with "rc-ignore-default-offer" tag
+     *   - Uses [SubscriptionOption] WITH longest free trial or cheapest first phase
+     *   - Falls back to use base plan
+     */
+    val defaultOffer: SubscriptionOption?
+        get() {
+            val basePlan = this.firstOrNull { it.isBasePlan } ?: return null
+
+            val validOffers = this
+                .filter { !it.isBasePlan }
+                .filter { !it.tags.contains("rc-ignore-default-offer") }
+
+            return findLongestFreeTrial(validOffers) ?: findLowestNonFreeOffer(validOffers) ?: basePlan
+        }
+
+    /**
+     * Finds all [SubscriptionOption]s with a specific tag.
```

Comment:
it might be helpful for us to call out that all offers inherit the base plan tag

Reply:
TIL 😅

Reply:
yeah... not my favorite fun fact..